File: 3026-find-the-minimum-possible-sum-of-a-beautiful-array/3026-find-the-minimum-possible-sum-of-a-beautiful-array.py
class Solution:
    def minimumPossibleSum(self, n: int, target: int) -> int:
        ans = 0
        st = set()
        temp = 1


        for i in range(1,n+1):
            while(temp in st):
                temp+=1
            ans += temp
            st.add(temp)
            st.add(target-temp)

        return ans

File: 3026-find-the-minimum-possible-sum-of-a-beautiful-array/3026-find-the-minimum-possible-sum-of-a-beautiful-array.cpp
class Solution {
public:
    long long minimumPossibleSum(int n, int target) {
        // return the three numbers 

        unordered_set<int> st;
        long long ans =0;

        int temp =1;
        for(int i =1;i<=n;i++){
            while(st.find(temp)!=st.end()){
                temp++;                
            }
            ans += temp;
            st.insert(temp);
            st.insert(target-temp);
            temp++;

        }








        return ans;


    }
};